etc/taskcluster/decision_task.py

```diff
@@ -0,0 +1,298 @@
+        echo "nightly" > rust-toolchain
+        ./mach build --dev
+        ./mach test-unit
+    """).create()
+
+
+def android_arm32():
+    return (
+        linux_build_task("Android ARMv7: build")
+        # file: NDK parses $(file $SHELL) to tell x64 host from x86
+        # wget: servo-media-gstreamer’s build script
+        .with_script("""
+            apt-get install -y --no-install-recommends openjdk-8-jdk-headless file wget
+            ./etc/ci/bootstrap-android-and-accept-licences.sh
+            ./mach build --android --release
+        """)
+        .with_artifacts(
+            "/repo/target/armv7-linux-androideabi/release/servoapp.apk",
+            "/repo/target/armv7-linux-androideabi/release/servoview.aar",
+        )
+        .find_or_create("build.android_armv7_release." + CONFIG.git_sha)
+    )
+
+
+def windows_dev():
+    return (
+        windows_build_task("Windows x64: dev build + unit tests")
+        .with_script(
+            # Not necessary as this would be done at the start of `build`,
+            # but this allows timing it separately.
+            "mach fetch",
+
+            "mach build --dev",
+            "mach test-unit",
+            "mach package --dev",
+        )
+        .with_artifacts("repo/target/debug/msi/Servo.exe",
+                        "repo/target/debug/msi/Servo.zip")
+        .find_or_create("build.windows_x64_dev." + CONFIG.git_sha)
+    )
+
+
+def windows_release():
+    return (
+        windows_build_task("Windows x64: release build")
+        .with_script("mach build --release",
+                     "mach package --release")
+        .with_artifacts("repo/target/release/msi/Servo.exe",
+                        "repo/target/release/msi/Servo.zip")
+        .find_or_create("build.windows_x64_release." + CONFIG.git_sha)
+    )
+
+
+def linux_wpt():
+    release_build_task = linux_release_build()
+    total_chunks = 2
+    for i in range(total_chunks):
+        this_chunk = i + 1
+        wpt_chunk(release_build_task, total_chunks, this_chunk, extra=(this_chunk == 1))
+
+
+def linux_release_build():
+    return (
+        linux_build_task("Linux x64: release build")
+        .with_script("""
+            ./mach build --release --with-debug-assertions -p servo
+            ./etc/ci/lockfile_changed.sh
+            tar -czf /target.tar.gz \
+                target/release/servo \
+                target/release/build/osmesa-src-*/output \
+                target/release/build/osmesa-src-*/out/lib/gallium
+        """)
+        .with_artifacts("/target.tar.gz")
+        .find_or_create("build.linux_x64_release." + CONFIG.git_sha)
+    )
+
+
+def wpt_chunk(release_build_task, total_chunks, this_chunk, extra):
+    name = "Linux x64: WPT chunk %s / %s" % (this_chunk, total_chunks)
+    script = """
+        ./mach test-wpt \
+            --release \
+            --processes 24 \
+            --total-chunks "$TOTAL_CHUNKS" \
```

Comment:
It’s less important in the case of WPT chunck number/total since the same info is repeated in the task name, but I find that env variable make task definitions easier to read. For example, you can find `"GIT_SHA": "6935a2ee19a1375f97100e9bd34848827156ba97"` in https://tools.taskcluster.net/groups/c9vIR8VYS1mk_6PNVGBgeA/tasks/bfxoMXMfTSatNSKWBlSxtQ/details without hunting for that info in the middle of a JSON-encoded multi-line bash script.